packages/dds/tree/src/simple-tree/schemaFactory.ts

```diff
@@ -339,6 +347,34 @@ export class SchemaFactory<
 		>;
 	}
 
+	/**
+	 * Ensures that the set of view keys in the schema is unique.
+	 * Also ensure that the final set of stored keys (including those implicitly derived from view keys) is unique.
+	 * @throws Throws a `UsageError` if either of the key uniqueness invariants is violated.
+	 */
+	private static assertUniqueKeys<
```

Comment:
I hadn't even though of adding this, but including it was a great call, and good job with the UsageErrors!